pkg/api/personal_token/personal_token.go

```diff
@@ -0,0 +1,66 @@
+package personal_token
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/aziontech/azion-cli/pkg/cmd/version"
+	"github.com/aziontech/azion-cli/pkg/logger"
+	"github.com/aziontech/azion-cli/utils"
+	"go.uber.org/zap"
+
+	sdk "github.com/aziontech/azionapi-go-sdk/personal_tokens"
+)
+
+type Client struct {
+	apiClient *sdk.APIClient
+}
+
+func NewClient(c *http.Client, url string, token string) *Client {
+	conf := sdk.NewConfiguration()
+	conf.HTTPClient = c
+	conf.AddDefaultHeader("Authorization", "token "+token)
+	conf.AddDefaultHeader("Accept", "application/json;version=3")
+	conf.UserAgent = "Azion_CLI/" + version.BinVersion
+	conf.Servers = sdk.ServerConfigurations{
+		{URL: url},
+	}
+	conf.HTTPClient.Timeout = 30 * time.Second
+
+	return &Client{
+		apiClient: sdk.NewAPIClient(conf),
+	}
+}
+
+type Response interface {
+	GetUuid() string
+	GetName() string
+	GetKey() string
+	GetUserId() float32
+	GetCreated() time.Time
+	GetExpiresAt() time.Time
+	GetDescription() string
+}
+
+type Request struct {
+	sdk.CreatePersonalTokenRequest
+}
+
+func (c *Client) Create(ctx context.Context, req *Request) (Response, error) {
+	logger.Debug("Create Personal Token")
+
+	response, httpResp, err := c.apiClient.PersonalTokenApi.CreatePersonalToken(ctx).
+		CreatePersonalTokenRequest(req.CreatePersonalTokenRequest).Execute()
+
+	if err != nil {
+		logger.Error("Error while creating a personal token", zap.Error(err))
+		logger.Debug("Status Code", zap.Any("http", httpResp.StatusCode))
+		logger.Debug("Headers", zap.Any("http", httpResp.Header))
+		logger.Debug("Response body", zap.Any("http", httpResp.Body))
```

Comment:
I was testing these right now. If, for example, there is a connection error and the requests is timed-out, these will be nil and will cause a panic.

Reply:
I don't understand

Reply:
checking if the httpResp actually has a response and place the logs from status to response body inside an if (checking that) would solve it, right? @PatrickMenoti @maxwelbm

Reply:
It would... but we have to think if it's worth doing this